### PR TITLE
CarAi now functional 

### PR DESCRIPTION
**Pull request by Tomerd-afk**
I added back in the Car Ai as well as the movement component abstract class. This time in one merge.

### CODE REVIEW

File: Assets/Scripts/AI/CarAi.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+
+    public override void Update()
+    {
+        //print(hp.HitPoints);
+        base.Update();
+    }
+
+    public float hitpoints
```
Comment (toryjleo) [inline Assets/Scripts/AI/CarAi.cs:38]: please move getter above methods

File: Assets/Scripts/AI/CarAi.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            //These functions determine if the car will back up or try to turn around
+            if (distanceToTarget < attackRange) 
+            {
+                movementComponent.control(dot, z);
+            } else if (distanceToTarget >attackRange)
+            {
+                movementComponent.control(1, z);
+            } 
+
+
```
Comment (toryjleo) [inline Assets/Scripts/AI/CarAi.cs:85]: unneeded newlines. cleanup

File: Assets/Scripts/AI/CarAi.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            //These functions determine if the car will back up or try to turn around
+            if (distanceToTarget < attackRange) 
+            {
+                movementComponent.control(dot, z);
+            } else if (distanceToTarget >attackRange)
+            {
+                movementComponent.control(1, z);
+            } 
+
```
Comment (LiamMillerSE) [inline Assets/Scripts/AI/CarAi.cs:84]: Looks good, just spacing format

**Review by toryjleo — CHANGES_REQUESTED** [review]

File: Assets/Scripts/AI/MovementComponent.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    protected float Acceleration
+    {
+        get
+        {
+           return Mathf.Clamp(300 / ACCELERATION_SCALE, STARTING_HEALTH / ACCELERATION_SCALE, MAX_ACCELERATION);
+        }
+    }
+    
+
+    // Number of player hit points
```
Comment (LiamMillerSE) [inline Assets/Scripts/AI/MovementComponent.cs:52]: Code missing or old comment

**Review by LiamMillerSE — APPROVED** [review]

**Comment by Tomerd-afk** [comment]
Compiled Changes

**Review by LiamMillerSE — APPROVED** [review]

**Comment by toryjleo** [comment]
:shipit:

**Pull request merged**